src/rgw/rgw_data_sync.h

```diff
@@ -597,6 +597,7 @@ class RGWDefaultSyncModule : public RGWSyncModule {
 class RGWArchiveSyncModule : public RGWDefaultSyncModule {
 public:
   RGWArchiveSyncModule() {}
+  bool supports_writes() override { return false; }
```

Comment:
It should be allowed writing in the archive zone if the user operates on the RGW(s) associated to the archive zone and it has valid permissions. It should returns true.